pkg/devfile/adapters/common/command.go

```diff
@@ -194,7 +194,7 @@ func ValidateCommand(data data.DevfileData, command common.DevfileCommand) (err
 
 	isComponentValid := false
 	for _, component := range components {
-		if command.Exec.Component == component.Container.Name {
+		if command.Exec.Component == component.Name {
```

Comment:
I'm sorry, I was just glancing through. But does this mean the above statement is true if i have a volume component whose name matches the command exec component?

Reply:
Because i was working on a clean up PR that was touching these files

Reply:
@maysunfaisal Because of the new format for component names, I don't believe it's valid any more to have two components of different types, but with the same name.

Reply:
@maysunfaisal As mentioned by @johnmcollier, I don't think it will be possible with the new schema.

Reply:
No i meant what if I have an invalid devfile like this:
```
components:
  - name: tools
    container:
      image: maysunfaisal/springbootbuild
      memoryLimit: 768Mi
      command: ['tail']
      args: [ '-f', '/dev/null']
      volumeMounts:
        - name: springbootpvc
          path: /data
      mountSources: true
  - name: springbootpvc
    volume: {}
commands:
  - id: defaultBuild
    exec:
      component: springbootpvc
      commandLine: "/artifacts/bin/build-container-full.sh"
      workingDir: /projects
      group:
        kind: build
        isDefault: true
```

but i realized that `component` is coming from `components := GetDevfileContainerComponents(data)`, so we're good :)